pkg/serviceaccount/externalsigner/v1beta1/service.proto

```diff
@@ -0,0 +1,60 @@
+syntax = "proto3";
+
+package v1beta1;
+
+service JWTKeyService {
+  // Sign an incoming payload
+  rpc SignPayload(SignPayloadRequest) returns (SignPayloadResponse) {}
+  // List supported Algorithms
+  rpc ListAlgorithms(ListAlgorithmsRequest) returns (ListAlgorithmsResponse) {}
+  // Get a specific key by ID, defaulting to the current public key
+  rpc GetPublicKey(GetPublicKeyRequest) returns (GetPublicKeyResponse) {}
+  // List all active public keys
+  rpc ListPublicKeys(ListPublicKeysRequest) returns (ListPublicKeysResponse) {}
+}
+
+message SignPayloadRequest {
+  // payload is the content to be signed
+  bytes payload = 1;
+  // algorithm specifies which algorithm to sign with
+  string algorithm = 2;
+}
+message SignPayloadResponse {
+  // content returns the signed payload
+  bytes content = 1;
+}
+
+message ListAlgorithmsRequest {}
+message ListAlgorithmsResponse {
+  // signature_algorithms returns the supported algorithms
+  repeated string signature_algorithms = 1;
+}
+
+message PublicKey {
+  // public_key is a PEM encoded public key
+  bytes public_key = 1;
+  // certificate is a concatenated list of PEM encoded x509 certificates
+  bytes certificates = 2;
+  // key_id is the key's ID
+  string key_id = 3;
+  // algorithm states the algorithm the key uses
```

Comment:
Can we whitelist the algorithms that we use today and add verification that we don't see anything unexpected?